### PR TITLE
Pick Ubuntu version and DIND

### PR DESCRIPTION
**Pull request by dagguh**

### CODE REVIEW

File: src/main/kotlin/com/atlassian/performance/tools/sshubuntu/api/SshUbuntuContainer.kt [hunk 1]
```diff
@@ -60,13 +57,31 @@ class SshUbuntuContainer(
      * https://github.com/testcontainers/testcontainers-java/issues/318
      * The class is a workaround for the problem.
      */
-    private class UbuntuContainer(version: String) : GenericContainer<UbuntuContainer>(
-        ImageFromDockerfile(/* dockerImageName = */ "ssh-ubuntu", /* deleteOnExit = */ false)
-            .withFileFromString("Dockerfile", SshUbuntuContainer::class.java.getResource("/docker/Dockerfile.template").readText().replace("%UBUNTU_VERSION%", version))
-            .withFileFromClasspath("authorized_keys", "/docker/authorized_keys")
-    ) {
```
Comment (dagguh) [inline src/main/kotlin/com/atlassian/performance/tools/sshubuntu/api/SshUbuntuContainer.kt:67]: This was failing on Ubuntu 22.04:
<details>
<summary><tt>shouldWorkOnJammy</tt> stacktrace</summary>
<pre>
net.schmizz.sshj.userauth.UserAuthException: Exhausted available authentication methods
	at net.schmizz.sshj.SSHClient.auth(SSHClient.java:231)
	at net.schmizz.sshj.SSHClient.authPublickey(SSHClient.java:346)
	at net.schmizz.sshj.SSHClient.authPublickey(SSHClient.java:395)
	at com.atlassian.performance.tools.ssh.api.auth.PublicKeyAuthentication.authenticate$ssh(PublicKeyAuthentication.kt:25)
	at com.atlassian.performance.tools.ssh.api.Ssh.prepareClient(Ssh.kt:97)
	at com.atlassian.performance.tools.ssh.api.Ssh.newConnection(Ssh.kt:43)
	at com.atlassian.performance.tools.sshubuntu.api.ssh.SshUbuntuTest.runSsh(SshUbuntuTest.kt:127)
	at com.atlassian.performance.tools.sshubuntu.api.ssh.SshUbuntuTest.testRelease(SshUbuntuTest.kt:94)
	at com.atlassian.performance.tools.sshubuntu.api.ssh.SshUbuntuTest.shouldWorkOnJammy(SshUbuntuTest.kt:90)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.runTestClass(JUnitTestClassExecutor.java:110)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.execute(JUnitTestClassExecutor.java:58)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.execute(JUnitTestClassExecutor.java:38)
	at org.gradle.api.internal.tasks.testing.junit.AbstractJUnitTestClassProcessor.processTestClass(AbstractJUnitTestClassProcessor.java:66)
	at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.processTestClass(SuiteTestClassProcessor.java:51)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:35)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:32)
	at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:93)
	at com.sun.proxy.$Proxy2.processTestClass(Unknown Source)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.processTestClass(TestWorker.java:118)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:35)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:175)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:157)
	at org.gradle.internal.remote.internal.hub.MessageHub$Handler.run(MessageHub.java:404)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:46)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
	at java.lang.Thread.run(Thread.java:750)
</pre>
</details>

**Review by mgrzaslewicz — APPROVED** [review]
It would be great to know the context (or ticket).
Why are we making this change?

File: CHANGELOG.md [hunk 2]
```diff
@@ -23,6 +23,13 @@ Dropping a requirement of a major version of a dependency is a new contract.
 ## [Unreleased]
 [Unreleased]: https://github.com/atlassian/ssh-ubuntu/compare/release-0.2.6...master
 
+### Added
+- Add `SshUbuntuContainer.Builder` with `version` override.
+- Add `SshUbuntuContainer.Builder.enableDocker`.
```
Comment (mgrzaslewicz) [inline CHANGELOG.md:28]: NIT: add more context. Why are we adding this?

**Comment by dagguh** [comment]
> Add `SshUbuntuContainer.Builder` with `version` override.

We're adding this, because there's more than one legit Ubuntu version.

> Add `SshUbuntuContainer.Builder.enableDocker`.

We're adding Docker support, because Docker is quite powerful. The evidence for that is the `ssh-ubuntu` lib itself, which relies on Docker. So in general that should be enough justification, but I can be more specific: it would allow reuse of existing Docker-backed APIs like [`MySqlDatabase`](https://github.com/atlassian/infrastructure/blob/release-4.20.0/src/main/kotlin/com/atlassian/performance/tools/infrastructure/api/database/MySqlDatabase.kt#L23-L26).

**Review by mgrzaslewicz — APPROVED** [review]

**Pull request merged**